packages/rum-core/src/common/utils.js

```diff
@@ -401,6 +401,36 @@ function isPerfTypeSupported(type) {
   )
 }
 
+/**
+ * The goal of this is to make sure that HAR files
+ * can be created containing beacons with the payload readable
+ */
+function isBeaconInspectionEnabled() {
+  const flagName = '_elastic_inspect_beacon_'
+
+  if (sessionStorage.getItem(flagName) !== null) {
+    return true
+  }
+
+  if (!window.URL || !window.URLSearchParams) {
+    return false
+  }
+
+  try {
+    const parsedUrl = new URL(window.location.href)
+    const isFlagSet = parsedUrl.searchParams.has(flagName)
+    if (isFlagSet) {
+      sessionStorage.setItem(flagName, '')
```

Comment:
maybe set to something to boolean?

Reply:
Yes I'm with you, even if the boolean is stored as a string, I believe it looks better than an empty string from the code standpoint.